CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/panes/DataAccessViewCategoryPanelController.java

```diff
@@ -96,8 +107,26 @@ public boolean isChanged() {
         final Preferences prefs = NbPreferences.forModule(DataAccessViewPreferenceKeys.class);
         final DataAccessViewCategoryPanel panel = getPanel();
         final List<String> hiddenCategory = panel.getHiddenCategory();
-        return (!hiddenCategory.isEmpty())
-                || (!prefs.get(DataAccessViewPreferenceKeys.HIDDEN_DA_VIEW, DataAccessViewPreferenceKeys.HIDDEN_DA_VIEW_DEFAULT).isEmpty());
+        final List<String> visibleCategory = panel.getVisibleCategory();
+
+        if (reorderButtonPressed && !visibleCategory.equals(visibleNow)) {
+            orderChanged = true;
+            visibleNow = panel.getVisibleCategory();
+        }
+
+        if (moveButtonPressed) {
+            firstLoad = false;
+        }
+
+        return visibleEntriesHaveChanged() || orderChanged;
+    }
+
+    private boolean visibleEntriesHaveChanged() {
+        if (firstLoad) {
+            return false;
+        }
+
+        return getPanel().getVisibleCategory().size() != visibleNow.size();
```

Comment:
Checking the size is not enough, you also need to check the entries individually.
You could have a visible list containing entries A, B, and C, and a hidden list containing entry D.
Then move entry D back to the visible list and move entry A to the hidden list.
So you now have B, C, and D in the visible list, and A in the hidden list.
The size has remained the same, but the list now has different entries, 
so the **visibleEntriesHaveChanged** function should be returning **true**.

Reply:
Yeah this was pretty dumb will fix it now.